subprojects/core/src/integTest/groovy/org/gradle/internal/scan/config/BuildScanAutoApplyIntegrationTest.groovy

```diff
@@ -0,0 +1,219 @@
+    }
+
+    def "does not automatically apply buildscan plugin when --scan is not provided on command-line"() {
+        when:
+        runBuildWithoutScanRequest()
+
+        then:
+        buildScanPluginNotApplied()
+    }
+
+    def "does not automatically apply buildscan plugin to subprojects"() {
+        when:
+        settingsFile << """
+            include 'a', 'b'
+"""
+        buildFile << """
+            assert pluginManager.hasPlugin('com.gradle.build-scan')
+            subprojects {
+                assert !pluginManager.hasPlugin('com.gradle.build-scan')
+            }
+"""
+
+        and:
+        runBuildWithScanRequest()
+
+        then:
+        buildScanPluginApplied(BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION)
+    }
+
+    def "does not apply buildscan plugin to buildSrc build"() {
+        when:
+        file('buildSrc/build.gradle') << """
+            println 'in buildSrc'
+            assert !pluginManager.hasPlugin('com.gradle.build-scan')
+"""
+
+        and:
+        runBuildWithScanRequest()
+
+        then:
+        outputContains 'in buildSrc'
+        buildScanPluginApplied(BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION)
+    }
+
+    def "does not apply buildscan plugin to nested builds in a composite"() {
+        when:
+        settingsFile << """
+            includeBuild 'a'
+"""
+        file('a/settings.gradle') << """
+            rootProject.name = 'a'
+"""
+        file('a/build.gradle') << """
+            println 'in nested build'
+            assert !pluginManager.hasPlugin('com.gradle.build-scan')
+"""
+
+        and:
+        runBuildWithScanRequest()
+
+        then:
+        outputContains 'in nested build'
+        buildScanPluginApplied(BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION)
+    }
+
+    @Unroll
+    def "uses #sequence version of plugin when explicit in plugins block"() {
+        when:
+        if (version != BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION) {
+            publishDummyBuildScanPlugin(version)
+        }
+        pluginsRequest "id 'com.gradle.build-scan' version '$version'"
+
+        and:
+        runBuildWithScanRequest()
+
+        then:
+        buildScanPluginApplied(version)
+
+        where:
+        sequence | version
+        "older"  | BUILD_SCAN_PLUGIN_MINIMUM_VERSION
+        "same"   | BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION
+        "newer"  | "100.0"
+    }
+
+    @Unroll
+    def "uses #sequence version of plugin when added to buildscript classpath"() {
+        when:
+        if (version != BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION) {
+            publishDummyBuildScanPlugin(version)
+        }
+        buildscriptApply "com.gradle:build-scan-plugin:$version"
+
+        and:
+        runBuildWithScanRequest()
+
+        then:
+        buildScanPluginApplied(version)
+
+        where:
+        sequence | version
+        "older"  | BUILD_SCAN_PLUGIN_MINIMUM_VERSION
+        "same"   | BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION
+        "newer"  | "100.0"
```

Comment:
:o: Should probably used `BuildScanPluginAutoApply.BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION` +1 to avoid future potential breakage.

Reply:
done